lib/Component/Available.php

```diff
@@ -13,6 +13,10 @@
  * @copyright Copyright (C) fruux GmbH (https://fruux.com/)
  * @author Ivan Enderlin
  * @license http://sabre.io/license/ Modified BSD License
+ *
+ * @property \DateTime $DTSTART
```

Comment:
I dont know very much about how this lib works, therefore reverse engineering the types based on the implmentation.

in case someone knows more then me and can help me to make the possible types stricter, feel free to jump in @evert @DominikTo @Hywan

Reply:
Is it possible to make this something like typescript's `any` ?

Reply:
not sure what `any` is in TS. but we can used `mixed` in case you want to allow everything.

Reply:
but our implementation of this class requires at least a object which support a `getDateTime()` method. everything else would create a runtime error in `getEffectiveStartEnd()`

Reply:
`mixed` sounds like the sane thing. I think having support for any property that can be any type might make this a lot simpler. Being pretty grounded into the Javasript / Typescript ecosystem now I don't think this style of Sabre\VObject design works well with static type analyzers. It's hard to be comprehensive because types can also be overridden with the `TYPE` parameter.

Reply:
having the properties defined as `mixed` would at least make the tools aware that those properties exist.

I dont get why mixed would be usefull here because of `getEffectiveStartEnd()` only working with a `getDateTime()` method though.
I expected it expects a DateTime object - but was just a guess

Reply:
It can almost expect any type. It's not dependent on the property name

Reply:
thats true for the base class `VObject\Component` but for the concrete subclasses we have a more concrete set of properties and values, no?

Reply:
As far as I know many properties can have a multitude of types, and even more types are accepted when setting (as opposed to getting) properties.

Reply:
The real issue I have is that the types are just very incomplete, which means that anyone using this library with phpstan or other phpdoc tools are going to get a barrage of errors even when they're using the library as intended.

So the types need to be either strict and exhaustive or not there at all.

Reply:
Hmm ok. So will change everything to mixed then

Reply:
I think that's the safest. Ideally also just assume that any property can exist =)